app/code/Magento/Wishlist/Model/ResourceModel/Item/Collection.php

```diff
@@ -292,7 +292,14 @@ protected function _assignProducts()
         $productCollection->addPriceData()
             ->addTaxPercents()
             ->addIdFilter($this->_productIds)
-            ->addAttributeToSelect($this->_wishlistConfig->getProductAttributes())
+            ->addAttributeToSelect($this->_wishlistConfig->getProductAttributes());
+
+        $this->_eventManager->dispatch(
```

Comment:
In this case it would be better to refactor logic for plugins usage. We don't recommend pass to event objects that should be modified in observers according to our [Technical Guidelines](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html#14-events).